pkg/util/containers/v2/metrics/docker/collector.go

```diff
@@ -81,13 +81,18 @@ func (d *dockerCollector) GetContainerStats(containerNS, containerID string, cac
 	}
 	outStats := convertContainerStats(&stats.Stats)
 
+	// Try to collect the container's PIDs via Docker API, if we can't spec() will fill in the entry PID
+	outStats.PID.PIDs, err = d.pids(containerID)
+	if err != nil {
+		log.Debugf("Unable to collect container's PIDs, PID list will be incomplete, cid: %s, err: %v", containerID, err)
```

Comment:
Are we expecting this to happen (often)?  If not, should probably be `warn` level (or maybe info) so that it would be captured in a flare in case of a support incident.

Reply:
No, we're not. Good call, adjusted.